Controller.java

```diff
@@ -7,58 +7,44 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/**
- * Servlet implementation class Controller
- */
+
 @WebServlet("/hello")
 public class Controller extends HttpServlet {
-
-	/**
-	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
-	 */
+	
+	protected void Writers (HttpServletRequest request, HttpServletResponse response, String fname) 
+			throws ServletException, IOException {
+		try (PrintWriter writer = response.getWriter()){
+			fname = request.getParameter("fname");
+			writer.println("nick: " + fname);
+	}
+}
 	protected void doGet(HttpServletRequest request, HttpServletResponse response) 
             throws ServletException, IOException {    
-
-        try (PrintWriter writer = response.getWriter()){
-            String nick = request.getParameter("nick");
-        	writer.println("Nickname: " + nick);
-        }
+        Writers(request, response, request.getParameter("nick: "));
     }
 
-	/**
-	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
-	 */
-	protected void doPost(HttpServletRequest req, HttpServletResponse res) 
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) 
             throws ServletException, IOException {
           
-        try (PrintWriter writer = res.getWriter()) {
-            String name = req.getParameter("username");
-            String age = req.getParameter("userage");
-            String gender = req.getParameter("gender");
-            String hobby = req.getParameter("hobby");
-            String[] visited_country = req.getParameterValues("visited_country");
-            
-            writer.println("<p>Name: " + name + "</p>");
-            writer.println("<p>Age: " + age + "</p>");
-            writer.println("<p>Gender: " + gender + "</p>");
-            writer.println("<p>Hobby: " + hobby + "</p>");
-            writer.println("<h3>country: </h3>");
+        try (PrintWriter writer = response.getWriter()) {
+        	Writers(request, response, request.getParameter("username: "));
+        	Writers(request, response, request.getParameter("userage: "));
+        	Writers(request, response, request.getParameter("gender: "));
+        	Writers(request, response, request.getParameter("hobby: "));
```

Comment:
Create the one string what you need to send and send it one time, not several times another strings